Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/TracingRepository.kt

```diff
@@ -121,59 +120,57 @@ class TracingRepository @Inject constructor(
     /**
      * Launches the RetrieveDiagnosisKeysTransaction and RiskLevelTransaction in the viewModel scope
      *
-     * @see RiskLevelTransaction
      * @see RiskLevelRepository
      */
     // TODO temp place, this needs to go somewhere better
     fun refreshRiskLevel() {
-        scope.launch {
-            try {
 
-                // get the current date and the date the diagnosis keys were fetched the last time
-                val currentDate = DateTime(Instant.now(), DateTimeZone.UTC)
-                val lastFetch = DateTime(
-                    LocalData.lastTimeDiagnosisKeysFromServerFetch(),
-                    DateTimeZone.UTC
+        // get the current date and the date the diagnosis keys were fetched the last time
+        val currentDate = DateTime(Instant.now(), DateTimeZone.UTC)
+        val lastFetch = DateTime(
+            LocalData.lastTimeDiagnosisKeysFromServerFetch(),
+            DateTimeZone.UTC
+        )
+
+        // check if the keys were not already retrieved today
+        val keysWereNotRetrievedToday =
+            LocalData.lastTimeDiagnosisKeysFromServerFetch() == null ||
+                currentDate.withTimeAtStartOfDay() != lastFetch.withTimeAtStartOfDay()
+
+        // check if the network is enabled to make the server fetch
+        val isNetworkEnabled =
+            ConnectivityHelper.isNetworkEnabled(CoronaWarnApplication.getAppContext())
+
+        // only fetch the diagnosis keys if background jobs are enabled, so that in manual
+        // model the keys are only fetched on button press of the user
+        val isBackgroundJobEnabled =
+            ConnectivityHelper.autoModeEnabled(CoronaWarnApplication.getAppContext())
+
+        Timber.tag(TAG)
+            .v("Keys were not retrieved today $keysWereNotRetrievedToday")
+        Timber.tag(TAG).v("Network is enabled $isNetworkEnabled")
+        Timber.tag(TAG)
+            .v("Background jobs are enabled $isBackgroundJobEnabled")
+
+        if (keysWereNotRetrievedToday && isNetworkEnabled && isBackgroundJobEnabled) {
+            // TODO shouldn't access this directly
```

Comment:
Here is a todo, should it be included in this change?

Reply:
That part still needs refactoring, @chris-cwa just moved it around I think, not in the scope of this PR I'd say.